test/users.js

```diff
@@ -0,0 +1,90 @@
+const chai = require('chai'),
+  chaiHttp = require('chai-http'),
+  dictum = require('dictum.js'),
+  server = require('../app'),
+  should = chai.should();
+
+chai.use(chaiHttp);
+describe('/users POST', () => {
+  it('should fail because email is missing', done => {
+    chai
+      .request(server)
+      .post('/users')
+      .send({
+        firstName: 'John',
+        lastName: 'Smith',
+        password: 'password'
+      })
+      .catch(err => {
+        err.should.have.status(400);
+        err.response.should.be.json;
+        err.response.body.should.have.property('message');
+        err.response.body.should.have.property('internal_code', 'invalid_email');
+      })
+      .then(() => done());
+  });
+  it('should fail because password does not meet requirements', done => {
+    chai
+      .request(server)
+      .post('/users')
+      .send({
+        firstName: 'John',
+        lastName: 'Smith',
+        email: 'john.smith@wolox.com.ar',
+        password: '@#'
+      })
+      .catch(err => {
+        err.should.have.status(400);
+        err.response.should.be.json;
+        err.response.body.should.have.property('message');
+        err.response.body.should.have.property('internal_code', 'saving_error');
+      })
+      .then(() => done());
+  });
+  it('should be succesful', done => {
+    chai
+      .request(server)
+      .post('/users')
+      .send({
+        firstName: 'John',
+        lastName: 'Smith',
+        email: 'john.smith@wolox.com.ar',
+        password: 'wolox1189'
+      })
+      .then(res => {
+        res.should.have.status(201);
```

Comment:
validate that it was actually created in the database, the controller could just be returning 201 without doing anything else and this test would still pass. Import the db model and fetch the use and check that the inputs values match with the database values, including the encrypted password

Reply:
Test now validates it was created correctly